Makefile_packaging.mk

```diff
@@ -0,0 +1,277 @@
+# Of course that does not work for SLES-12
+ID := $(shell . /etc/os-release; echo $$ID)
+VERSION_ID := $(shell . /etc/os-release; echo $$VERSION_ID)
+ifeq ($(findstring opensuse,$(ID)),opensuse)
+ID_LIKE := suse
+DISTRO_ID := sl$(VERSION_ID)
+endif
+ifeq ($(ID),sles)
+# SLES-12 or 15 detected.
+ID_LIKE := suse
+DISTRO_ID := sle$(VERSION_ID)
+endif
+
+# For some reason this is needed for building some packages as while these
+# repos are present in zypp:// build is not finding packages in them.
+# libfabric needs libpsm2 from OpensUSE Leap 42.3 science:HPC repo.
+OSUSE_REPO := https://download.opensuse.org/repositories
+ifeq ($(DISTRO_ID),sle12.3)
+BUILD_OPTIONS += --repo
+BUILD_OPTIONS += $(OSUSE_REPO)/science:/HPC/openSUSE_Leap_42.3
+BUILD_OPTIONS += --repo
+BUILD_OPTIONS += http://cobbler/cobbler/repo_mirror/updates-sles12.3-x86_64
+BUILD_OPTIONS += --repo
+BUILD_OPTIONS += http://cobbler/cobbler/repo_mirror/sdkupdate-sles12.3-x86_64
+BUILD_OPTIONS += --repo
+BUILD_OPTIONS += http://cobbler/cobbler/repo_mirror/sdk-sles12.3-x86_64
+endif
+ifeq ($(DISTRO_ID),sl42.3)
+BUILD_OPTIONS += --repo
+BUILD_OPTIONS += $(OSUSE_REPO)/science:/HPC/openSUSE_Leap_42.3
+endif
+
+COMMON_RPM_ARGS := --define "%_topdir $$PWD/_topdir"
+DIST    := $(shell rpm $(COMMON_RPM_ARGS) --eval %{?dist})
+ifeq ($(DIST),)
+SED_EXPR := 1p
+else
+SED_EXPR := 1s/$(DIST)//p
+endif
+SPEC    := $(NAME).spec
+ifeq ($(VERSION),)
+VERSION := $(shell rpm $(COMMON_RPM_ARGS) --specfile --qf '%{version}\n' $(SPEC) | sed -n '1p')
+endif
+DOT     := .
+DEB_VERS := $(subst rc,~rc,$(VERSION))
+DEB_RVERS := $(subst $(DOT),\$(DOT),$(DEB_VERS))
+DEB_BVERS := $(basename $(subst ~rc,$(DOT)rc,$(DEB_VERS)))
+RELEASE := $(shell rpm $(COMMON_RPM_ARGS) --specfile --qf '%{release}\n' $(SPEC) | sed -n '$(SED_EXPR)')
+SRPM    := _topdir/SRPMS/$(NAME)-$(VERSION)-$(RELEASE)$(DIST).src.rpm
+RPMS    := $(addsuffix .rpm,$(addprefix _topdir/RPMS/x86_64/,$(shell rpm --specfile $(SPEC))))
+DEB_TOP := _topdir/BUILD
+DEB_BUILD := $(DEB_TOP)/$(NAME)-$(DEB_VERS)
+DEB_TARBASE := $(DEB_TOP)/$(DEB_NAME)_$(DEB_VERS)
+SOURCES := $(addprefix _topdir/SOURCES/,$(notdir $(SOURCE)) $(PATCHES))
+ifeq ($(ID_LIKE),debian)
+DEBS    := $(addsuffix _$(DEB_VERS)-1_amd64.deb,$(shell sed -n '/-udeb/b; s,^Package:[[:blank:]],$(DEB_TOP)/,p' debian/control))
+CLEAN_TARBALS := $(shell rm -f *.tar.$(SRC_EXT))
+#Ubuntu Containers do not set a UTF-8 environment by default.
+ifndef LANG
+export LANG = C.UTF-8
+endif
+ifndef LC_ALL
+export LC_ALL = C.UTF-8
+endif
+TARGETS := $(DEBS)
+else
+# CentOS/Suse packages that want a locale set need this.
+ifndef LANG
+export LANG = en_US.utf8
+endif
+ifndef LC_ALL
+export LC_ALL = en_US.utf8
+endif
+TARGETS := $(RPMS) $(SRPM)
+endif
+all: $(TARGETS)
+
+%/:
+	mkdir -p $@
+
+_topdir/SOURCES/%: % | _topdir/SOURCES/
+	rm -f $@
+	ln $< $@
+
+# At least one spec file, SLURM (sles), has a different version for the
+# download file than the version in the spec file.
```

Comment:
:confounded:  This is just so very wrong.  The spec, and thus the resulting binary RPMs **really** should reflect what's in the spec.  Upstream using `-` in their version string is anti-packaging.

Reply:
This was probably for a  case where we only need to make Debian packages, so no spec file would be present.
I will remove it until we identify if we really have that case.